File: backend/src/models/Payment.ts
import { getModelForClass, modelOptions, prop } from "@typegoose/typegoose";

enum States {
  paid = "Paid",
  pending = "Pending",
  delayed = "Delayed",
}

@modelOptions({
  schemaOptions: {
    timestamps: true,
  },
})
export class Payment {
  @prop({ type: Number, required: true })
  amount: number;
  @prop({ type: String, required: true, enum: States, default: "Pending" })
  state: string;
  @prop({ type: Date, required: true })
  payment_date: Date;
  @prop({ type: Number })
  days_late: number;
  @prop({ type: String })
  receipt: string;
}

const PaymentModel = getModelForClass(Payment);
export default PaymentModel;

File: frontend/lib/models/Payment.dart
class Payment {
  final String state, paymentDate, receipt;
  Payment({
    required this.state,
    required this.paymentDate,
    required this.receipt,
  });
}

List<Payment> payments = [
  Payment(
      state: "Paid",
      paymentDate: "2022/05/15",
      receipt:
          "http://192.168.0.32:8000/receipts/6330651f771aac756ffdb621/2022-9-25.pdf"),
  Payment(
      state: "Paid",
      paymentDate: "2022/05/17",
      receipt:
          "http://192.168.0.32:8000/receipts/6330651f771aac756ffdb621/2022-9-25.pdf"),
  Payment(
      state: "Paid",
      paymentDate: "2022/05/20",
      receipt:
          "http://192.168.0.32:8000/receipts/6330651f771aac756ffdb621/2022-9-25.pdf")
];
